resources/NpmDsc/NpmDsc.psm1

```diff
@@ -1,12 +1,199 @@
+        [string]$Command
+    )
+
+    return Invoke-Expression -Command "npm $Command"
+}
+
+function Set-PackageDirectory {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$PackageDirectory
+    )
+
+    if (Test-Path -Path $PackageDirectory -PathType Container) {
+        Set-Location -Path $PackageDirectory
+    } else {
+        throw "$($PackageDirectory) does not point to a valid directory."
+    }
+}
+
+function Get-InstalledNpmPackages {
+    param (
+        [Parameter()]
+        [bool]$Global
+    )
+
+    $command = [List[string]]::new()
+    $command.Add('list')
+    $command.Add('--json')
+
+    if ($Global) {
+        $command.Add('-g')
+    }
+
+    return Invoke-Npm -Command $command
+}
+
+function Install-NpmPackage {
+    param (
+        [Parameter()]
+        [string]$PackageName,
+
+        [Parameter()]
+        [bool]$Global,
+
+        [Parameter()]
+        [string]$Arguments
+    )
+
+    $command = [List[string]]::new()
+    $command.Add('install')
+    $command.Add($PackageName)
+
+    if ($Global) {
+        $command.Add('-g')
+    }
+
+    $command.Add($Arguments)
+
+    Write-Verbose -Message "Executing 'npm $command'"
+
+    return Invoke-Npm -Command $command
+}
+
+function Uninstall-NpmPackage {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$PackageName,
+
+        [Parameter()]
+        [bool]$Global,
+
+        [Parameter()]
+        [string]$Arguments
+    )
+
+    $command = [List[string]]::new()
+    $command.Add('uninstall')
+    $command.Add($PackageName)
+
+    if ($Global) {
+        $command.Add('-g')
+    }
+
+    $command.Add($Arguments)
+
+    Write-Verbose -Message "Executing 'npm $command'"
+
+    return Invoke-Npm -Command $command
+}
+
+function GetNpmPath {
+    $npmPath = if ($IsWindows) {
+        $npmCacheDir = Join-Path $env:LOCALAPPDATA 'npm-cache' '_logs'
+        $globalNpmCacheDir = Join-Path $env:SystemDrive 'npm' 'cache' '_logs'
+        if (Test-Path $npmCacheDir -ErrorAction SilentlyContinue) {
+            $npmCacheDir
+        }
+        elseif (Test-Path $globalNpmCacheDir -ErrorAction SilentlyContinue) {
+            $globalNpmCacheDir
+        } else {
+            $result = (Invoke-Npm -Command 'config list --json' | ConvertFrom-Json -ErrorAction SilentlyContinue).cache
+            if (Test-Path $result -ErrorAction SilentlyContinue) {
+                $result
+            }
+            else {
+                return $null
+            }
+        }
+    }
+    elseif ($IsLinux -or $IsMacOS) {
+        $npmCacheDir = Join-Path $env:HOME '.npm/_logs'
+        if (Test-Path $npmCacheDir -ErrorAction SilentlyContinue) {
+            $npmCacheDir
+        }
+        else {
+            return $null
+        }
+    }
+    else {
+        throw 'Unsupported platform'
+    }
+
+    return $npmPath
+}
+
+function GetNpmWhatIfResponse {
+    $npmPath = GetNpmPath
+    $errorMessages = if ($null -ne $npmPath) {

```

Comment:
Same here, if you don't plan on using $errorMessages later on in this function, just return the value directly.